src/store/actions.ts

```diff
@@ -0,0 +1,80 @@
+import { v4 } from 'uuid';
+
+import { addNoteToLS, getNotes, removeNoteFromLS, setEditedNotesInLS } from 'api/localStorage';
+import { addNote, deleteNoteInState, resetIsNotesFetching, setIsNotesFetching, setNotes } from 'store/notes-slice';
+import { Note } from 'store/notes-slice.types';
+import { AppDispatch } from 'store/store';
+
+export const fetchNotes = () => async (dispatch: AppDispatch) => {
+  const notes = getNotes();
+  dispatch(setIsNotesFetching());
+  dispatch(setNotes(notes));
+  dispatch(resetIsNotesFetching());
+};
+
+export const setNewNote = (values: Note) => async (dispatch: AppDispatch) => {
+  dispatch(setIsNotesFetching());
+  const note = addNoteToLS(values);
+  if (note) {
+    dispatch(addNote(note));
+  }
+  dispatch(resetIsNotesFetching());
+};
+
+export const removeNote = (values: string) => async (dispatch: AppDispatch) => {
+  const noteId = removeNoteFromLS(values);
+  if (noteId === 'ok') {
+    dispatch(deleteNoteInState(values));
+  }
+};
+
+export const editNote = (values: Note) => async (dispatch: AppDispatch) => {
+  let notes = getNotes();
+  let UpdatingNote = notes.find((items: Note) => items.id === values.id);
```

Comment:
1. it should be camelCase naming
2.  try to use `notes.map(...)` instead of `find()` and `filter()`
3.  move the edit logic to `setEditedNotesInLS`
4. rename `setEditedNotesInLS` into `editeNoteById`